src/connection.js

```diff
@@ -82,23 +52,20 @@ class Connection extends EventEmitter {
     }
 
     if (typeof config.server !== 'string') {
-      throw new TypeError('Invalid server: ' + config.server);
+      throw new TypeError('The "config.server" property is required and must be of type string.');
     }
 
-    if (config.domain != undefined) {
-      deprecateNonStringConfigValue('domain', config.domain);
+    if (config.domain !== undefined && typeof config.domain !== 'string') {
+      throw new TypeError('The "config.domain" property must be of type string.');
     }
-    deprecateNullConfigValue('domain', config.domain);
 
-    if (config.userName != undefined) {
-      deprecateNonStringConfigValue('userName', config.userName);
+    if (config.userName !== undefined && typeof config.userName !== 'string') {
+      throw new TypeError('The "config.userName" property must be of type string.');
```

Comment:
Shouldn't all these conditions be this `||` instead of `&&`?

Reply:
I think this check is right. 🤔 `config.userName`, `config.password` and `config.domain` are all optional and thus can be `undefined`, or  need to be a string.